### DISCUSSION TITLE
¿What may I use to evaluate the quality of the net?

### DISCUSSION THREAD

**Discussion by Jumargo** [openthread/openthread#7985] [Q&A] [answered]
My sistem is based in:
- 1 RCP ot-ctl device (efr32MG13 chip)
- 50 NCP FTD without CLI (nrf52840 chip)

I have found this:
https://openthread.io/codelabs/openthread-testing-visualization#0

That example is based on 6 DevKit nrf52840 but I only have 1 DevKit. Is it enough? When we install the Silk we have to modify the hwconfig.ini file so I have no clear if I have to fill the file with 1 + 50 devices properties (impossible because my devices have no J-Link onboard for looking the DutSerial) or is enought with the info of my DevKit.

If this is not possible, how could I test my OT net? Maybe I have to reflash my 50 NCP FTD with CLI?

Regards,

**Comment by jwhui** [comment, marked as answer]
It really depends on how you define "quality". As with any evaluation of technology, there are many metrics that may be used to measure "quality". If you want to measure something like latency or throughput across a fixed topology, then a tool like ping or iperf may be useful. If you want to observe behavior across a variety of different network topologies, then you will likely need some tool to help configure those network topologies.

Separately, if you are trying to leverage an existing test framework such as Silk, then you will need to support the HW requirements of those frameworks. Silk does leverage JTAG to put firmware on devices.

There may not be an existing framework that satisfies your goals. In that case, you may need to look at building your own test framework.
